lib/containers/utils.pm

```diff
@@ -315,4 +315,78 @@ sub reset_container_network_if_needed {
+
+    # Test --volumes-from
+    assert_script_run("$runtime run -d --name test test");
+    assert_script_run("$runtime run --rm --volumes-from test test ls /test | grep test");
+    assert_script_run("$runtime rm -vf test");
+
+    assert_script_run("$runtime run -d --name test -v \$PWD/test:/test test");
+    # NOTE: When https://github.com/containers/podman/issues/19529 is solved we must add a version check here
+    my $ret = script_run("$runtime run --rm --volumes-from test test ls /test/test");
+    record_soft_failure("gh#19529 - Unexpected error with --volumes-from") if ($ret != 0 && $runtime == "podman");
+    assert_script_run("$runtime rm -vf test");
+
+    # Test --volume option with directory (read-only)
+    assert_script_run("! $runtime run --rm --volume \$PWD/test:/test:ro test rm /test/test");
+    assert_script_run("test -f test/test");
+
+    # Equivalent --mount option to above
+    assert_script_run("! $runtime run --rm --mount type=bind,source=\$PWD/test,destination=/test,readonly test rm /test/test");
+    assert_script_run("test -f test/test");
+
+    assert_script_run("rm -f test/test");
+
+    # Test --volume option with directory (read-write)
+    assert_script_run("$runtime run --rm --volume \$PWD/test:/test test touch /test/test");
+    assert_script_run("test -f test/test");
+
+    assert_script_run("rm -f test/test");
+
+    # Equivalent --mount option to above
+    assert_script_run("$runtime run --rm --mount type=bind,source=\$PWD/test,destination=/test test touch /test/test");
+    assert_script_run("test -f test/test");
+
+    assert_script_run("rm -f test");
+
+    # Test volume subcommands
+    # Subcommands supported only by podman: exists export import mount reload unmount
+    assert_script_run("$runtime volume create test");
+    assert_script_run("$runtime volume ls --format '{{.Name}}' | grep -Fx test");
+    assert_script_run("$runtime volume inspect test | jq '.[0].Name' | grep -Fx '\"test\"'");
+    assert_script_run("$runtime volume rm test");
+
+    assert_script_run("$runtime volume create test");
+
+    # Test --volume option with volume (read-write)
+    assert_script_run("$runtime run --rm --volume test:/test test touch /test/test");
+    assert_script_run("$runtime run --rm --volume test:/test test ls /test/test");
+
+    # Equivalent --mount option to above
+    assert_script_run("$runtime run --rm --mount type=volume,source=test,destination=/test test touch /test/test2");
+    assert_script_run("$runtime run --rm --mount type=volume,source=test,destination=/test test ls /test/test2");
+
+    # Test --volume option with volume (read-only)
+    assert_script_run("! $runtime run --rm --volume test:/test:ro test rm /test/test");
+
+    # Equivalent --mount option to above
+    assert_script_run("! $runtime run --rm --mount type=volume,source=test,destination=/test,readonly test rm /test/test");
+
+    assert_script_run("$runtime volume prune -f | grep -Fx test");
```

Comment:
Good idea! Perhaps we would like to check afterwards also if the dangling volumes got deleted?

Reply:
I believe it would be better to use (maybe extending would be unavoidable though) https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/engine.pm#L322

Reply:
Fixed.

Reply:
> I believe it would be better to use (maybe extending would be unavoidable though) https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/engine.pm#L322

I extended that function but I would want to specifically test the `volume` subcommand here.

Reply:
> Good idea! Perhaps we would like to check afterwards also if the dangling volumes got deleted?

Fixed.